### PR TITLE
Add default InfoFormat selection to WMS Catalog 

### PR DESCRIPTION
**Pull request by ashraf-alsamman**
## Description

first part: get info format from service and make a select stage and save to service 
second part: make info format option selected by default

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [X] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#6840

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: build/tests.webpack.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
+var context = require.context('../web/client/components/catalog/editor/__tests__', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```
Comment (offtherailz) [inline build/tests.webpack.js:1]: You should not commit this
```suggestion
var context = require.context('../web', true, /(-test\.jsx?)|(-test-chrome\.jsx?)$/);
```

File: web/client/utils/CatalogUtils.js [hunk 2]
```diff
@@ -694,6 +694,26 @@ export const DEFAULT_FORMAT_WMS = [{
...
+//     label: "TEXT",
+//     value: "TEXT",
+//     format: "text/plain"
+// }, {
+//     label: "PROPERTIES",
+//     value: "PROPERTIES",
+//     format: "application/json"
+// }];
+
+
```
Comment (offtherailz) [inline web/client/utils/CatalogUtils.js:716]: Please do not commit commented code 

```suggestion
```

File: web/client/components/TOC/fragments/settings/Display.jsx [hunk 3]
```diff
@@ -18,14 +18,34 @@ import InfoPopover from '../../../widgets/widget/InfoPopover';
...
+}, {
+    label: "TEXT",
+    value: "TEXT",
+    format: "text/plain"
+}, {
+    label: "PROPERTIES",
+    value: "PROPERTIES",
+    format: "application/json"
+}];
+
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/settings/Display.jsx:40]: You should use the Feature Info tab. 
```suggestion
import { DEFAULT_FORMAT_WMS, getSupportedFormat } from '../../../../utils/CatalogUtils';
```

File: web/client/components/TOC/fragments/settings/Display.jsx [hunk 4]
```diff
@@ -18,14 +18,34 @@ import InfoPopover from '../../../widgets/widget/InfoPopover';
...
 export default class extends React.Component {
     static propTypes = {
         opacityText: PropTypes.node,
         element: PropTypes.object,
         formats: PropTypes.array,
         settings: PropTypes.object,
         onChange: PropTypes.func,
+        getUniqueInfoFormats: PropTypes.func,
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/settings/Display.jsx:48]:
```suggestion
```

File: web/client/components/TOC/fragments/settings/Display.jsx [hunk 5]
```diff
@@ -112,6 +132,8 @@ export default class extends React.Component {
         }
         return null;
     };
+
+
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/settings/Display.jsx:136]: Don't add unuseful changes
```suggestion
```

File: web/client/components/TOC/fragments/settings/Display.jsx [hunk 6]
```diff
@@ -165,6 +187,8 @@ export default class extends React.Component {
                                 }}/>
                         </FormGroup>
                     </Col>
+
+
```
Comment (offtherailz) [inline web/client/components/TOC/fragments/settings/Display.jsx:191]: Don't add unuseful changes

**Review by offtherailz — CHANGES_REQUESTED** [review]
Here a first list of things to clean up. 
I wasn't able to test the effective task, but I only reviewed the code finding things that have to be cleaned up or fixed for sure. 
Please 
-[ ] See and fix accordingly to my comments inline
-[ ] make the actions build (it means that translations for it,en,de,us,fr have to be added  and eslint stuff have to be fixed, tests should pass).

File: web/client/components/TOC/fragments/settings/Display.jsx (hunk 6 above)
Comment (ashraf-alsamman) [reply web/client/components/TOC/fragments/settings/Display.jsx:191]: Of course, thanks for informing me

File: web/client/utils/CatalogUtils.js (hunk 2 above)
Comment (ashraf-alsamman) [reply web/client/utils/CatalogUtils.js]: ok ,

File: web/client/translations/data.en-US.json [hunk 7]
```diff
@@ -149,6 +149,7 @@
                 "tooltip": "Note: This parameter requires specific configurations on GeoServer"
             },
             "format": {
+                "infoFormat": "Info format",
```
[the remark this answers is not in the document]
Comment (ashraf-alsamman) [reply web/client/translations/data.en-US.json:152]: Sure, i will do

File: web/client/components/TOC/fragments/settings/Display.jsx (hunk 5 above)
Comment (ashraf-alsamman) [reply web/client/components/TOC/fragments/settings/Display.jsx:136]: sure

File: web/client/translations/data.hr-HR.json [hunk 8]
```diff
@@ -103,6 +103,17 @@
...
+            "infoFormat": "Info format",
+            "title": "Format",
+            "refresh": "Dohvati podržani format",
+            "loading": "Učitavanje ...",
+            "noOption": "Nema opcije",
+            "error": {
+                "title": "Pogreška",
+                "message": "Dohvaćanje formata nije uspjelo"
+            }
+        },
```
Comment (offtherailz) [inline web/client/translations/data.hr-HR.json:116]: This is not needed, and anyway is not alligned with other translations. 
```suggestion
```

File: web/client/translations/data.fi-FI.json [hunk 9]
```diff
@@ -102,7 +101,18 @@
...
+        "infoFormat": "Info-muoto",
+        "title": "Muoto",
+        "refresh": "Hae tuettu muoto",
+        "loading": "Ladataan ...",
+        "noOption": "Ei vaihtoehtoa",
+        "error": {
+            "title": "Virhe",
+            "message": "Muoto noutaminen epäonnistui"
+        }
+    }
```
Comment (offtherailz) [inline web/client/translations/data.fi-FI.json:115]: not needed, not aligned with the others

File: web/client/translations/data.pt-PT.json [hunk 10]
```diff
@@ -103,6 +103,17 @@
...
+            "Format": {
+                "infoFormat": "Information format",
+                "title": "Format",
+                "update": "Search for a compatible format",
+                "loading": "Loading...",
+                "noOption": "No option",
+                "error": {
+                    "title": "Error",
+                    "message": "Failed to fetch format"
+                }
```
Comment (offtherailz) [inline web/client/translations/data.pt-PT.json:116]: Not needed, not aligned with the other and also the camel case is wrong.
```suggestion
```

File: web/client/translations/data.nl-NL.json [hunk 11]
```diff
@@ -104,7 +103,7 @@
             "templateFormatInfoAlert2": "Gebruik ${ attribute } om de eigenschappen die je wenst te tonen aan te duiden.",
             "templateFormatInfoAlertExample": "Het id van het object is ${ properties }",
             "templatePreview": "Sjabloon voorbeeld",
-	    "visibilityLimits": {
+	        "visibilityLimits": {
```
Comment (offtherailz) [inline web/client/translations/data.nl-NL.json:106]: please do not send changes that are not necessary or errors, check your diffs before.

File: web/client/components/catalog/editor/AdvancedSettings/RasterAdvancedSettings.js [hunk 12]
```diff
@@ -10,6 +10,29 @@ import { FormGroup, Col, ControlLabel } from "react-bootstrap";
...
 import { DEFAULT_FORMAT_WMS } from '../../../../utils/CatalogUtils';
 import localizedProps from '../../../misc/enhancers/localizedProps';
+import Message from '../../../I18N/Message';
+
+// TODO: import this from the proper file (the original has something like {HTML: "text/html"})
+
+const DEFAULT_INFO_FORMATS = [{
+    label: "DISABLED",
+    value: "HTML",
+    format: "text/html"
```
Comment (offtherailz) [inline web/client/components/catalog/editor/AdvancedSettings/RasterAdvancedSettings.js:20]: I think this causes the lissue of disabled.

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] Here the label is wrong. It was "Format"m now became "About", please make it "Format" or "Image format", to make it more understandable.
- [ ] Formats in the selector should be localized in the same way we localize the titles in display card. See `defaultSettingsTabs` --> `formatCards`
image
- [ ] The option "DISABLED" doesn't work, it makes it "HTML". The correct value  should be HIDDEN. Please verify by checking the effective state used when this option is selected from layer settings
disabled
- Tested with this server `https://demo.mapserver.org/cgi-bin/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetCapabilities` that doesn't support html or properties, I still see the HTML option, even if not selectable.
image
- [ ] Comments inline (things committed not necessary)
- [ ] Formats in Display tab should be filtered by "infoFormats". (2nd part of the issue - reporting for completeness.

File: web/client/components/catalog/Catalog.jsx [hunk 13]
```diff
@@ -271,7 +272,10 @@ class Catalog extends React.Component {
                 onAddBackground={this.props.onAddBackground}
                 defaultFormat={this.props.services[this.props.selectedService] && this.props.services[this.props.selectedService].format}
                 formatOptions={this.props.services[this.props.selectedService]?.url === this.props.service?.url ? this.props.formatOptions : DEFAULT_FORMAT_WMS}
-                infoFormatOptions={this.props.services[this.props.selectedService]?.url === this.props.service?.url ? this.props.infoFormatOptions : getUniqueInfoFormats()}
+                // infoFormatOptions={this.props.services[this.props.selectedService]?.url === this.props.service?.url ? this.props.infoFormatOptions : getUniqueInfoFormats()}
```
Comment (offtherailz) [inline web/client/components/catalog/Catalog.jsx:275]: Do not commit commented code
```suggestion
```

File: web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx [hunk 14]
```diff
@@ -64,10 +64,7 @@ describe('Test CatalogServiceEditor', () => {
             const labels = [...fg.querySelectorAll('label')];
             return labels.length === 1 && labels[0].textContent === 'Format';
         });
-        expect(formatFormGroups.length).toBe(1);
-        const formatSelect = formatFormGroups[0].querySelector('.Select-value-label');
-        expect(formatSelect).toExist();
-        expect(formatSelect.textContent).toBe('image/png8');
```
Comment (offtherailz) [inline web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx:70]: These checks are still valid.

File: web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx [hunk 15]
```diff
@@ -97,10 +94,7 @@ describe('Test CatalogServiceEditor', () => {
             const labels = [...fg.querySelectorAll('label')];
             return labels.length === 1 && labels[0].textContent === 'WMS Layer tile size';
         });
-        expect(wmstileSizeFormGroups.length).toBe(1);
-        const tileSizeSelect = wmstileSizeFormGroups[0].querySelector('.Select-value-label');
-        expect(tileSizeSelect).toExist();
-        expect(tileSizeSelect.textContent).toBe("256x256");
```
Comment (offtherailz) [inline web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx:103]: These checks are still valud

File: web/client/translations/data.de-DE.json [hunk 16]
```diff
@@ -149,16 +149,17 @@
...
             },
             "format": {
-                "title": "Format",
-                "refresh": "Unterstütztes Format abrufen",
-                "loading": "Wird geladen...",
-                "noOption": "Keine Option",
-                "error": {
-                    "title": "Error",
-                    "message": "Format konnte nicht abgerufen werden"
-                }
```
Comment (offtherailz) [inline web/client/translations/data.de-DE.json:159]: Do not change indentation or add lines to i18n files

File: web/client/translations/data.es-ES.json [hunk 17]
```diff
@@ -159,6 +159,7 @@
                 }
             }
         },
+
```
Comment (offtherailz) [inline web/client/translations/data.es-ES.json:162]:
```suggestion
```

File: web/client/translations/data.nl-NL.json [hunk 18]
```diff
@@ -137,6 +136,17 @@
...
+                "infoFormat": "Info-muoto",
+                "title": "Muoto",
+                "refresh": "Hae tuettu muoto",
+                "loading": "Ladataan ...",
+                "noOption": "Ei vaihtoehtoa",
+                "error": {
+                    "title": "Virhe",
+                    "message": "Muoto noutaminen epäonnistui"
+                }
+            },
```
Comment (offtherailz) [inline web/client/translations/data.nl-NL.json:149]:
```suggestion
```

File: web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx (hunk 15 above)
Comment (ashraf-alsamman) [reply web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx:103]: not working with i18n

Comment (offtherailz) [reply web/client/components/catalog/__tests__/CatalogServiceEditor-test.jsx:103]: I think that only

```
onst wmstileSizeFormGroups = [...document.querySelectorAll('.form-group')].filter(fg => {
            const labels = [...fg.querySelectorAll('label')];
            return labels.length === 1 && labels[0].textContent === 'WMS Layer tile size';
        });
        expect(wmstileSizeFormGroups.length).toBe(1);
        expect(wmstileSizeFormGroups.length).toBe(1);
```

Have to be replaced with something different. The rest of the test is ok.

You may find the proper label/select by getting the new not localized string (what "textContent" contains now?") or by assigning a class to it.

**Review by offtherailz — CHANGES_REQUESTED** [review]
I take this note only for future changes to this pending work.

We should revert to the previous review and fix what was not working, that was not that much.
Here we have several problems:
- General Tab il layer settings has been replaced with Dispay tab, that changes should be reverted
- The dropdown for formats in catalog now 
        - The disabled/hidden entry as well as loading finishes
        - Shows template, that should not be shown.

- Other details: 
   - We have several formatting changes committed that are not needed.
   - The list should not show before loading capabilities is completed. the default lists should be shown only if we don't have the remote data.
    - Need to resolve conflicts (General settings changes ,that need anyway to be reverted)

**Comment by tdipisa** [comment]
@offtherailz what should we do with this PR?

**Comment by offtherailz** [comment]
This work need to be finalized and fixed  as for the review. 
If you prefer we can close it, taking note that partial work is on this branch.

**Comment by offtherailz** [comment]
Going to close, will open another PR, branch already on upstream branch

**Pull request closed without merging**